Set Support system_libs if WIN32, not just MSVC or MINGW

z2oh opened the pull request:
The previous check was false when compiling with `clang++`, which prevented `ntdll` from being specified as a link library, causing an undefined symbol error when trying to resolve `RtlGetLastNtStatus`. Since we always want to link these libraries on Windows, the check can be simplified to just `if( WIN32 )`.

z2oh committed:
Set Support system_libs if WIN32, not just MSVC or MINGW

The previous check was false when compiling with clang++, which prevented ntdll from being specified as a link library, causing an undefined symbol error when trying to resolve RtlGetLastNtStatus. Since we always want these libs on Windows, the check can be simplified to just if(WIN32).

mstorsjo reviewed (approved):
LGTM, thanks!

Note that the PR description (the first post) is what ends up as the actual commit message, when merged with "squash and merge" (although it's possible to edit it when merging). So it can be worthwhile to polish it up and split out parts that aren't intended as the final commit message (e.g. the references to me) ;-)

z2oh commented:
Ah I didn't know that! I assumed it took the commit's message. I've removed the extra comments from the first post, and pasted them below for posterity:
> I added the reference to RtlGetLastNtStatus in https://github.com/llvm/llvm-project/pull/90655.
> 
> I tried building with clang++ without this change, saw the undefined symbol error, then tried building again with if( WIN32 ) here and was able to build successfully.
> 
> Thanks @mstorsjo for suggesting this straightforward fix.
